File: 响应式/ReactiveCocoa/ReactiveCocoa/Classes/Others/Category/UIButton+Extension.m
//
//  UIButton+Extension.m
//  ReactiveCocoa
//
//  Created by 金亮齐 on 2017/10/11.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "UIButton+Extension.h"

@implementation UIButton (Extension)
- (void)title:(NSString *)title titleColor:(UIColor *)titleColor image:(NSString *)image backgroundColor:(UIColor *)backgroundColor fontSize:(CGFloat)fontSize target:(id)target action:(SEL)action{
    
    [self setTitle:title forState:UIControlStateNormal];
    [self setTitleColor:titleColor forState:UIControlStateNormal];
    self.backgroundColor = backgroundColor;
    self.titleLabel.font = [UIFont systemFontOfSize:fontSize];
    [self addTarget:target action:action forControlEvents:UIControlEventTouchUpInside];
    self.titleEdgeInsets = UIEdgeInsetsMake(0, 10, 0, 0);
    [self setImage:[UIImage imageNamed:image] forState:UIControlStateNormal];
}
@end

File: Example/Swift项目/花田小憩/花田小憩/Classes/Tools/Category/UIButton+Extension.swift

//
//  UIButton+Extension.swift
//  花田小憩
//
//  Created by 金亮齐 on 2016/12/21.
//  Copyright © 2016年 醉看红尘这场梦. All rights reserved.
//

import UIKit

extension UIButton
{
    convenience init(title: String?, imageName: String?, target: AnyObject? ,selector: Selector?, font: UIFont?, titleColor: UIColor?) {
        self.init()
        if let imageN = imageName {
            setImage(UIImage(named:imageN), forState: .Normal)
        }
        setTitleColor(titleColor, forState: .Normal)
        titleLabel?.font = font
        setTitle(title, forState: .Normal)
        if let sel = selector {
            addTarget(target, action: sel, forControlEvents: .TouchUpInside)
        }
        
    }
}
